src/lib_ccx/matroska.c

```diff
@@ -814,6 +823,9 @@ void save_sub_track(struct matroska_ctx* mkv_ctx, struct matroska_sub_track* tra
             write(desc, timestamp_end, strlen(timestamp_start));
             write(desc, ",", 1);
             char* text = ass_ssa_sentence_erase_read_order(sentence->text);
+            while(text[0]=='\n' || text[0]=='\N'){
```

Comment:
ASS/SSA line ending is two chars. Check that text[0] is '\' and text[1] is 'n' or 'N'